crates/subspace-farmer/src/plot.rs

```diff
@@ -79,11 +82,29 @@ struct RequestWithPriority {
 
 struct Inner {
     handlers: Handlers,
-    requests_sender: Option<mpsc::SyncSender<RequestWithPriority>>,
-    plot_metadata_db: Option<Arc<DB>>,
+    requests_sender: mpsc::SyncSender<RequestWithPriority>,
+    plot_metadata_db: Arc<DB>,
     piece_count: Arc<AtomicU64>,
 }
 
+impl Drop for Inner {
+    fn drop(&mut self) {
+        let (result_sender, result_receiver) = mpsc::channel();
+
+        if self
+            .requests_sender
+            .send(RequestWithPriority {
+                request: Request::Exit { result_sender },
+                priority: RequestPriority::Low,
+            })
```

Comment:
Would it be better to make it lowest in order to serve last read-requests?

Reply:
It is already the lowest priority. In practice though, this will only run if there is no Plot instances, so no requests will be in progress anyway.